### PR TITLE
Improve BBTools-related workflows

### PR DESCRIPTION
**Pull request by fanninpm**
- [x] This comment contains a description of what is in the pull request.

Following up from #304, this PR adds a deployment workflow for the BBTools image. In creating that workflow, I realized that I could deduplicate some code to make it easier to upgrade workflows in the future. Accordingly, I extracted every instance of `bbtools` and `38.96` in the workflow and put them in environment variables (declared in an `env` block). Hopefully, the "Deduplicate values in testing workflow" commit will make it easier for someone to follow along in case they would want to make similar changes to other workflows.

- [x] Update relevant GitHub actions workflow files
- [x] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
- [x] Have successfully run the workflow "Test <program name> image" in your forked repository

### CODE REVIEW

**Comment by fanninpm** [comment]
This ended up being a little bit more convoluted than I previously though. @erinyoung, what do you think?

File: .github/workflows/test-bbtools.yml [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+env:
+  TOOL_NAME: bbtools
+  TOOL_VERSION: "38.96"
+
 on:
   workflow_dispatch:
   pull_request:
     paths:
-      - 'bbtools/38.96/Dockerfile'
+      - 'bbtools/38.96/Dockerfile' # cannot use env vars
```
Comment (erinyoung) [inline .github/workflows/test-bbtools.yml:11]: Does this still work with 
```
- 'bbtools/*/Dockerfile'
```

Comment (fanninpm) [reply .github/workflows/test-bbtools.yml:11]: As far as I know, that would still work, but it would apply to _all_ versions of the tool.

**Comment by erinyoung** [comment]
My apologies. I wanted to add this to several other files. I'll time for it eventually. 

In the meantime, I'm going to merge this.

**Comment by kapsakcj** [comment]
I think it would be a good idea to update the deploy template `.github/workflow-templates/deploy-template.yml` if these changes are working as expected

**Pull request merged**